irr/src/OpenGL/VBO.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2024 sfan5
+// This file is part of the "Irrlicht Engine".
+// For conditions of distribution and use, see copyright notice in irrlicht.h
+
+#include "VBO.h"
+
+#include <cassert>
+#include <mt_opengl.h>
+
+namespace irr
+{
+namespace video
+{
+
+void OpenGLVBO::upload(const void *data, size_t size, size_t offset,
+		GLenum usage, bool mustShrink)
+{
+	bool newBuffer = false;
+	if (!m_name) {
+		GL.GenBuffers(1, &m_name);
+		if (!m_name)
+			return;
+		newBuffer = true;
+	} else if (size > m_size || mustShrink) {
+		// note: mustShrink && offset > 0 is forbidden
```

Comment:
How about adding an assertion in that case?

Reply:
will do